### PR TITLE
[RESTEASY-1628]

### PR DESCRIPTION
**Pull request by ronsigal**
Changed treatment of text media types without charset parameter to add
"charset=UTF-8" by default.

### CODE REVIEW

File: resteasy-jaxrs/src/main/java/org/jboss/resteasy/plugins/interceptors/MessageSanitizerContainerResponseFilter.java [hunk 1]
```diff
@@ -74,5 +76,18 @@ private String escapeXml(String str) {
         return sb.toString();
     }
 
+    private boolean containsHtmlText(ArrayList<Object> list) {
+       for (Iterator<Object> it = list.iterator(); it.hasNext();  ) {
+          Object o = it.next();
+          if (o instanceof String) {
+             String mediaType = (String) o;
+             String[] parts = mediaType.split("/");
+             if (parts.length >= 2 && parts[0].trim().equals("text") && parts[1].trim().startsWith("html")) {
```
Comment (kanovotn) [inline resteasy-jaxrs/src/main/java/org/jboss/resteasy/plugins/interceptors/MessageSanitizerContainerResponseFilter.java]: Just a question:
 * I haven't try it myself, but what if the content-type will be "TEXT/HTML", then containsHtmlText will return false and then the string in the entity (L36) will not be escaped? Other lines in this PR, who test content-type, use equalsIgnoreCase() method.

Comment (ronsigal) [inline resteasy-jaxrs/src/main/java/org/jboss/resteasy/plugins/interceptors/MessageSanitizerContainerResponseFilter.java]: You're so right, Katka. Thanks. I'll fix that.

Comment (ronsigal) [inline resteasy-jaxrs/src/main/java/org/jboss/resteasy/plugins/interceptors/MessageSanitizerContainerResponseFilter.java]: Changed to

          if (o instanceof String) {
             String mediaType = (String) o;
             String[] partsType = mediaType.split("/");
             if (partsType.length >= 2) {
                String[] partsSubtype = partsType[1].split(";");
                if (partsType[0].trim().equalsIgnoreCase("text") && 
                      partsSubtype[0].trim().toLowerCase().equals("html")) {
                   return true;
                }
             }
          }

Comment (kanovotn) [inline resteasy-jaxrs/src/main/java/org/jboss/resteasy/plugins/interceptors/MessageSanitizerContainerResponseFilter.java]: :+1:

**Comment by asoldano** [comment]
@rsvoboda , thanks for the review. I agree on the need for squashing the commits; we could move the containsHtmlText method to MediaTypeHelper (in that case it should probably be a more general-use one, like containsContentType(List<Object> contentTypes, String contentType), but that's not strictly needed. I agree on using "for (Object o : list)", while I think we could avoid the JDK8 syntax for now to have the same code in 3.0 branch. Finally, it's better not using MediaType.valueOf(o.toString()) as long as there's no real need of it, basically because building MediaType objects is quite heavy (they always come with a TreeMap in them and the valueOf method is quite complex).

**Comment by ronsigal** [comment]
- I squashed the commits, but, since master had changed, I had to create a new branch and a new PR: https://github.com/resteasy/Resteasy/pull/1110 . Do we have a policy restricting PRs to a single commit?
- I changed to the "for (Object o : list)" syntax.
- While creating a general purpose containsContentType() might be useful, I would have to do more work to parse the parameters, and probably wouldn't even want to use it in this case. So i left that alone.

**Comment by asoldano** [comment]
@ronsigal , thanks. No we don't have a policy for single commit PR in RESTEasy, but single commit changes for a given jira are usually better for future investigation of changes ;-) This said, it often happens to have many commits to fix a given jira..

**Comment by ronsigal** [comment]
@asoldano I definitely agree that 80 commits is too many. ;-)

**Pull request closed without merging**